libr/bin/format/pyc/marshal.c

```diff
@@ -1139,19 +1140,15 @@ static bool extract_sections_symbols(pyc_object *obj, RList *sections, RList *sy
 	RListIter *i = NULL;
 
 	//each code object is a section
-	if (!obj || (obj->type != TYPE_CODE_v1 && obj->type != TYPE_CODE_v0)) {
-		return false;
-	}
+	if_true_return (!obj || (obj->type != TYPE_CODE_v1 && obj->type != TYPE_CODE_v0), false);
```

Comment:
wasn't this suposed to be a r_return_val_if_fail?

Reply:
no. i decided to not use them because they can be disabled completely and these checks must be in place.

Reply:
to be clear, this conditions is needed. so i do not want to use something that can be removed at compilation time due macros.

```
#if R_CHECKS_LEVEL == 0

#define r_return_if_fail(expr) do { ; } while(0)
#define r_return_val_if_fail(expr, val) do { ; } while(0)
```

Reply:
absolutely! r_return_ are supposed to be used for checking programmatic errors (e.g. misuing an API, or things like that). In this case it seems more a runtime error, so it is something that should always be properly checked!

Reply:
well, what i dont like about this approach is that this `if_true_return` thing is defined locally here, where in the rest of the codebase we use the multiline ifs. im happy to introduce this macro r2-wide, but not like this https://github.com/radareorg/radare2/pull/17199/files#diff-b1ff7ca2270551aa1b19a80b3116c91bR9

i'm fine to merge this too. because this pr fixes several more things